gateway-ha/src/main/java/io/trino/gateway/ha/security/LbLdapClient.java

```diff
@@ -42,7 +47,9 @@ public UserEntryMapper(String memberOfAttr) {
 
     @Override
     public UserRecord map(Entry entry) throws LdapException {
-      return new UserRecord(entry.get(memberOf).toString());
+      Attribute attribute = entry.get(memberOf);
+      return ObjectUtils.isEmpty(attribute) ? emptyRecord : new UserRecord(attribute.getString());
```

Comment:
Is it possible to change the signature to `Optional<UserRecord>` and return `Optional.empty()` if there is no `memberOf` attribute? `Optional.empty()` is the usual way this type of situation is handled in Trino.

Reply:
Nope. 
![image](https://github.com/trinodb/trino-gateway/assets/25147023/3b100838-1767-4638-bede-dea0de35eecb)

Map function is used in two places below and cannot be empty :( 
- [return entry == null ? null : entryMapper.map( entry );](https://github.com/apache/directory-ldap-api/blob/2.1.2/ldap/client/api/src/main/java/org/apache/directory/ldap/client/template/LdapConnectionTemplate.java#L831)
- [entries.add( entryMapper.map( entry ) );](https://github.com/apache/directory-ldap-api/blob/2.1.2/ldap/client/api/src/main/java/org/apache/directory/ldap/client/template/LdapConnectionTemplate.java#L354)

Well it is possible to simply return a `new UserRecord(null)` instead though. What do you think?

Reply:
Sorry I should have been more clear. If you change 
```
public static class UserEntryMapper implements EntryMapper<UserRecord>
```
to
```
public static class UserEntryMapper implements EntryMapper<Optional<UserRecord>>
```
then make the necessary changes to other parts of the code to use `Optional`, then I _think_ this should all work.

Reply:
Oh my bad. I understood it wrong xD. Changed as you said and rebased it too!